Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -3011,6 +3011,41 @@ internal static bool ShouldDrawWarningIconForBinding(string bindingPath)
 
 #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
 
+#if UNITY_EDITOR
+        [InitializeOnLoad]
+        public static class PlayStateNotifier
+        {
+            static PlayStateNotifier()
+            {
+                EditorApplication.playModeStateChanged += ModeChanged;
+            }
+
+            static void ModeChanged(PlayModeStateChange playModeState)
+            {
+                // EnteredEditMode  Occurs during the next update of the Editor application if it is in edit mode and was previously in play mode.
+                // ExitingEditMode  Occurs when exiting edit mode, before the Editor is in play mode.
+                // EnteredPlayMode  Occurs during the next update of the Editor application if it is in play mode and was previously in edit mode.
+                // ExitingPlayMode  Occurs when exiting play mode, before the Editor is in edit mode.
+                switch (playModeState)
+                {
+                    case PlayModeStateChange.EnteredEditMode:
+                        actions?.Disable();
```

Comment:
Was there a reason for using this rather than ExitingPlayMode?

Reply:
Just wondering if there might be any implications of doing it after the mode change "edge"?

Reply:
I tried both ways and didn't see any difference.

It wasn't immediately clear to me if one was preferable to the other.

Reply:
This should ideally be covered in tests and I will try to cover that as part of modifying the tests which I currently do on a pending PR. Will resolve this.